core/vm/environment.go

```diff
@@ -73,6 +73,8 @@ type Environment interface {
 	DelegateCall(me ContractRef, addr common.Address, data []byte, gas, price *big.Int) ([]byte, error)
 	// Create a new contract
 	Create(me ContractRef, data []byte, gas, price, value *big.Int) ([]byte, common.Address, error)
+
+	StructLogs() []StructLog
```

Comment:
No, the reason is that ethapi.Backend uses the vm.Environment interface instead of *core.VMEnv (needs to be replaceable, there is a light implementation for VMEnv) and TraceCall needs to access StructLogs.